src/activation.rs

```diff
@@ -199,6 +193,20 @@ pub async fn get_activation_env<'p>(
         .collect())
 }
 
+/// Determine the environment variables that need to be set in an interactive shell to make it
+/// function as if the environment has been activated. This method runs the activation scripts from
+/// the environment and stores the environment variables it added, finally it adds environment
+/// variables from the project.
+pub async fn get_activation_env<'p>(
+    environment: &'p Environment<'p>,
+    lock_file_usage: LockFileUsage,
+) -> miette::Result<HashMap<String, String>> {
+    // Get the prefix which we can then activate.
+    get_up_to_date_prefix(environment, lock_file_usage, false, IndexMap::default()).await?;
```

Comment:
Isn't something like this already done as a prerequisite for the pypi resolve?

Reply:
this is the old part of existing get_activation_env. I've extracted the part what I need in get_env_and_activation_variables, so I could call it without causing infinite recursion